code/game/objects/items/storage/storage.dm

```diff
@@ -383,69 +385,98 @@
 
 	return TRUE
 
+/**
+ * This proc handles the delay associated with a storage object.
+ * If there is no delay, or the delay is negative, it simply returns TRUE.
+ * Should return true if the access delay is completed successfully.
+ */
+/obj/item/storage/proc/handle_access_delay(obj/item/accessed, mob/user, taking_out = TRUE, alert_user = TRUE)
+	if(!access_delay || !should_access_delay(accessed, user, taking_out))
+		return TRUE
+
+	if(!alert_user)
+		return do_after(user, access_delay, TRUE, src, ignore_turf_checks=TRUE)
+
+	to_chat(user, "<span class='notice'>You begin to [taking_out ? "take" : "put"] [accessed] [taking_out ? "out of" : "into"] [src]")
+	if(!do_after(user, access_delay, TRUE, src, ignore_turf_checks=TRUE))
+		to_chat(user, "<span class='warning'>You fumble [accessed]!</span>")
+		return FALSE
+	return TRUE
+
+/**
+ * This proc checks to see if we should actually delay access in this scenario
+ * This proc should return TRUE or FALSE
+ */
+/obj/item/storage/proc/should_access_delay(obj/item/accessed, mob/user, taking_out)
+	return FALSE
+
 //This proc handles items being inserted. It does not perform any checks of whether an item can or can't be inserted. That's done by can_be_inserted()
 //The stop_warning parameter will stop the insertion message from being displayed. It is intended for cases where you are inserting multiple items at once,
 //such as when picking up all the items on a tile with one click.
 //user can be null, it refers to the potential mob doing the insertion.
```

Comment:
```suggestion
/**
 * This proc handles items being inserted. It does not perform any checks of whether an item can or can't be inserted.     
 * That's done by can_be_inserted()
 * The stop_warning parameter will stop the insertion message from being displayed. It is intended for cases where you are inserting multiple items at once,
 * such as when picking up all the items on a tile with one click.
 * user can be null, it refers to the potential mob doing the insertion.
 */
```

Reply:
I'd rather not clean up existing code in this PR. If you want I can make another PR to clean up the entire storage module instead

Reply:
@SplinterGP

Reply:
Just press commit suggestion, its quick. Letting things ror later is never a good idea when you can just do it now

Reply:
hhhhh